### ISSUE TITLE
Migrate to stylelint 16

### ISSUE THREAD

**Issue by rollecode** [ronilaukkarinen/stylelint-a11y#4]
As stated in [stylelint-a11y/#65](https://github.com/YozhikM/stylelint-a11y/issues/65), stylelint 16 support is currently missing. Upgrading from 15 to 16 is not as simple as updating the package and peerDependencies, we also need to go through every single rule and upgrade them and tests to the new syntax.

I currently have no spare time for this work so I'm going to open this as issue.

PRs appreciated, will probably take a look at this some time in the beginning of 2024 after my well deserved holiday.

[Migration guide from Stylelint 15 to 16](https://stylelint.io/migration-guide/to-16/).

> This release contains significant or breaking changes.
> 
> We've migrated our source code to [ECMAScript modules](https://nodejs.org/docs/latest/api/esm.html) (ESM) — a year-long effort to allow ESM plugins, custom syntaxes and formatters, and a step towards updating our pure ESM dependencies.
> 
> To give the community time to migrate to ESM, we'll publish a hybrid package to support the (now deprecated) CommonJS Node.js API until our next major release.

**Comment by carlosjeurissen** [comment]
For stylelint 16 support, ESM is not yet needed.

One of the issues is the removal of some of the utils from stylelint 16. We will have to clone the missing ones from stylelint 15.

Another issue which will then arise is the following error: "stylelint requires plugins to expose a ruleName."

**Comment by carlosjeurissen** [comment]
Seems https://github.com/double-great/stylelint-a11y is actively maintained and has stylelint 16 support. So lets move our efforts to the double-great fork.

@ronilaukkarinen/stylelint-a11y can be the stylelint 15 compatible one for those who can't make the switch at this moment.

**Comment by rollecode** [comment]
> @ronilaukkarinen/stylelint-a11y can be the stylelint 15 compatible one for those who can't make the switch at this moment.

This is cool with me. Didn't know that compatible version exists. Thanks, stylelint 16 users can use that one. I'm going to close this issue.
